cpp/src/arrow/compute/compute-test.cc

```diff
@@ -670,6 +671,15 @@ TEST_F(TestCast, FromNull) {
   ASSERT_ARRAYS_EQUAL(*result, *result);
 }
 
+TEST_F(TestCast, EmptyArray) {
+  const int length = 0;
+  auto out_data = ArrayData::Make(int32(), length, {nullptr, nullptr});
+
+  auto arr = out_data.get();
+  auto ret = GetValues<int32_t>(*arr, 1);
+  ASSERT_EQ(ret, nullptr);
+}
```

Comment:
@Ghatage this test is not quite sufficient to test what we want to test which is that having a null buffer does not cause `compute::Cast` to segfault